dome9/data_source_dome9_awp_aws_get_onboarding_data.go

```diff
@@ -0,0 +1,93 @@
+package dome9
+
+import (
+	"encoding/base64"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwpAwsOnboardingData() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwpAwsOnboardingDataRead,
+
+		Schema: map[string]*schema.Schema{
+			"cloud_account_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"stage": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cloud_guard_backend_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"agentless_bucket_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"remote_functions_prefix_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"remote_snapshots_utils_function_name": {
```

Comment:
since when build the cft, we generate these variables, all of them need to be get from the api, since if these constants changed in awp agentless build process, we will also get the same change (so we decided to get all these values from the get onboarding data api)

Reply:
@bzemms we need to keep in mind backwards compatibility -- customers might have existing deployments of these terraform resources, so we can't make breaking changes without careful consideration..

Reply:
our lambda function. we can't really change its name or other parameters (in the future) - without worrying about backward compatibility with deployments before that future change. in fact, I see benefits in hard-coding it in the module rather than retrieving from backend. At least you can know which customer uses which version.